app/src/main/java/com/team7/cmput301/android/theirisproject/activity/ProblemListActivity.java

```diff
@@ -38,6 +40,16 @@ protected void onCreate(Bundle savedInstanceState) {
         problemsView = findViewById(R.id.problem_item_list);
         controller = createController(getIntent());
 
+        problemsView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+                Problem problem = (Problem) problemsView.getItemAtPosition(i);
+                Intent intent = new Intent(ProblemListActivity.this, ViewProblemActivity.class);
+                intent.putExtra("problem_id", problem.getId());
```

Comment:
Make problem_id a constant (EXTRA_PROBLEM_ID) in ViewProblemActivity so that it's more maintainable and you can use it here and in AddProblemActivity
Also, it should be prefaced by our domain name in case other applications use the same extra ID e.g.
"com.team7.cmput301.android.theirisproject.problem_id"